netket/vqs/mc/mc_mixed_state/expect_grad.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+from typing import Any, Callable, Tuple
+
+import numpy as np
+
+import jax
+from jax import numpy as jnp
+
+from netket import jax as nkjax
+from netket import config
+from netket.stats import Stats, statistics, mean
+from netket.utils import mpi
+from netket.utils.types import PyTree
+from netket.utils.dispatch import dispatch, TrueT, FalseT
+
+from netket.operator import (
+    DiscreteOperator,
+    AbstractSuperOperator,
+    local_cost_function,
+    local_value_cost,
+    Squared,
+    _der_local_values_jax,
+)
+
+from netket.vqs.mc import kernels, check_hilbert, get_configs, get_fun
+from .state import MCMixedState
+
+# mixed state, squared super-operator
+@dispatch
+def expect_and_grad(  # noqa: F811
```

Comment:
In other words, taking this function as an example, do I understand correctly that you plan to remove this from here and just make it a super general "empty shell" function that internally calls the relevant dispatched kernel?